File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int pre_sum=0, count=0;
        unordered_map<int,int> mp;
        mp[0]=1;
        for(int i=0;i<nums.size();i++)
        {
            pre_sum += nums[i];

            int x = mp[pre_sum - k] ;
            count+=x;
            
            mp[pre_sum]++;
        }
        return count;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefixSum = {}
        prefixSum[0] = 1
        sum = 0
        res = 0
        for i in range(len(nums)):

            sum += nums[i]

            diff = sum - k

            if diff in prefixSum:
                res += prefixSum[diff]
            
            if sum in prefixSum:
                prefixSum[sum] += 1
            else:
                prefixSum[sum] = 1
        
        return res